docs/fassets/developer-guides/12-fassets-autoredeem.mdx

```diff
@@ -0,0 +1,649 @@
+2. **Extracts Parameters**: Decodes the compose message to get:
+   - Amount to redeem (in lots)
+   - Underlying address (XRP address)
+   - Redeemer address (EVM address)
+3. **Gets Asset Manager**: Retrieves the FAsset AssetManager from Flare's ContractRegistry
+4. **Calculates Lots**: Determines how many lots can be redeemed based on received balance
+5. **Approves Tokens**: Grants AssetManager permission to spend the FAsset tokens
+6. **Executes Redemption**: Calls `assetManager.redeem()` to burn FAssets and release underlying XRP
+
+### Key Features
+
+- **Automatic Execution**: No manual redemption step required
+- **Reentrancy Protected**: Uses OpenZeppelin's `ReentrancyGuard`
+- **Owner Controls**: Allows owner to recover stuck tokens or native currency
+- **Event Emission**: Emits `RedemptionTriggered` event for tracking
+
+### Error Handling
+
+- `OnlyEndpoint()`: Reverts if caller is not the LayerZero endpoint
+- `InsufficientBalance()`: Reverts if no tokens were received
+- `AmountTooSmall()`: Reverts if received amount is less than 1 lot
+
+<details>
+<summary>View `FAssetRedeemComposer.sol` source code</summary>
+
+<CodeBlock language="solidity" title="contracts/FAssetRedeemComposer.sol">
+  {FassetRedeemComposer}
+</CodeBlock>
+
+</details>
+
+### Deployment
+
+Deploy the composer contract to Coston2:
+
+```bash
+# Set deployment configuration
+export DEPLOYER_PRIVATE_KEY=your_private_key
+
+# Deploy via Hardhat deploy scripts
+npx hardhat deploy --network coston2 --tags FAssetRedeemComposer
+
+# Save the deployed address to your .env
+export COSTON2_COMPOSER=0x...
+```
+
+---
+
+## Bridge FXRP to Hyperliquid EVM (Step 1)
+
+### Script: `bridgeToHyperEVM.ts`
+
+### What It Is
+
+**This is Step 1 of the two-step process** and is a **prerequisite** for testing the auto-redemption feature.
+
+This script bridges FXRP tokens from Coston2 to Hyperliquid EVM Testnet using LayerZero's OFT Adapter pattern.
+It wraps existing ERC20 FAsset tokens into LayerZero OFT format for cross-chain transfer.
+
+**Important:** This script does NOT perform auto-redemption. Its purpose is to get your FXRP tokens onto Hyperliquid EVM Testnet so that:
+
+- You can use them for trading or DeFi on Hyperliquid, OR
+- You can run the auto-redeem script (Step 2) to convert them back to native XRP
+
+You must successfully complete this step before running `autoRedeemHyperEVM.ts`.
+
+### How It Works
+
+#### Step-by-Step Process
+
+1. **Account Setup**: Retrieves the signer account from Web3
+2. **Balance Check**: Verifies user has sufficient FTestXRP tokens
+3. **Token Approval**:
+   - Approves OFT Adapter to spend FTestXRP (2x amount for safety)
+   - Approves Composer (if needed for future operations)
+4. **Build Send Parameters**:
+   - Destination: Hyperliquid EVM Testnet (EID: `40294`)
+   - Recipient: Same address on destination chain
+   - Amount: Configured bridge amount (default 11 FXRP)
```

Comment:
Ideally we should calculate this dynamically because it could change (not likely) but we can't have fixed lot size values. Or to push it forward explain what is this value **11**.